hazelcast/test/src/sql_test.cpp

```diff
@@ -505,6 +866,392 @@ TEST_F(SqlTest, statement_with_params)
+          return t.height == height_param || t.age == age_param;
+      });
+
+    auto result = client.get_sql().execute(statement).get();
+
+    for_each_row(result,
+                 assert_row_count(expected_row_counts),
+                 [&expecteds](const sql::sql_page::sql_row& row) {
+                     auto idx = row.get_object<int>(0);
+                     EXPECT_TRUE(idx.has_value());
+
+                     const auto& expected = expecteds.at(*idx);
+
+                     auto age = row.get_object<int64_t>(1);
+                     EXPECT_TRUE(age.has_value());
+                     EXPECT_EQ(*age, expected.age);
+
+                     auto height = row.get_object<float>(2);
+                     EXPECT_TRUE(height.has_value());
+                     EXPECT_EQ(row.get_object<float>(2), expected.height);
+                 });
+}
+
+TEST_F(SqlTest, test_execute_statement_with_params_after_clear_parameters)
+{
+    create_mapping_for_student();
+
+    auto expecteds = populate_map<test::student>(20);
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT __key, age, height "
+                       "FROM %1% "
+                       "WHERE height = CAST(? AS REAL) OR age = ?") %
+         map_name)
+          .str()
+    };
+
+    auto height_param = expecteds.front().height;
+    auto age_param = expecteds.back().age;
+
+    statement.set_parameters(height_param);
+    statement.clear_parameters();
+    statement.set_parameters(height_param, age_param);
+
+    ASSERT_NO_THROW(client.get_sql().execute(statement).get());
+}
+
+TEST_F(SqlTest, test_execute_statement_with_mismatched_params_when_sql_has_more)
+{
+    create_mapping();
+    (void)populate_map();
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT * FROM %1% WHERE __key > ? AND this > ?") %
+         map_name)
+          .str()
+    };
+
+    statement.set_parameters(5);
+    auto execution = client.get_sql().execute(statement);
+
+    EXPECT_THROW(execution.get(),
+                 hazelcast::client::sql::hazelcast_sql_exception);
+}
+
+TEST_F(SqlTest,
+       test_execute_statement_with_mismatched_params_when_params_has_more)
+{
+    create_mapping();
+    (void)populate_map();
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT * FROM %1% WHERE this > ?") % map_name).str()
+    };
+
+    statement.set_parameters(5, 6);
+    auto execution = client.get_sql().execute(statement);
+
+    EXPECT_THROW(execution.get(), sql::hazelcast_sql_exception);
+}
+
+TEST_F(SqlTest, test_execute_statement_with_timeout)
+{
+    create_mapping_for_student();
+    auto expecteds = populate_map<test::student>(100);
+
+    sql::sql_statement statement{
+        client,
+        (boost::format("SELECT __key, height FROM %1% WHERE height < 100") %
+         map_name)
+          .str()
+    };
+
+    statement.timeout(std::chrono::milliseconds{ 100 });
```

Comment:
Isn't this too low timeout which can be affected if the machine is slow to respond due to other running jobs?

Reply:
I suspected same thing but I copied timeout from python so I thought it is not a problem. I will make it `5000`